pkg/daemon/gpsd.go

```diff
@@ -251,13 +251,17 @@ retry:
 					} else if strings.Contains(output, "UBX-NAV-TIMELS") {
 						emptyCount = 0
 						missedTickers = 0
-						lines := []string{}
+						var lines []string
 						for i := 0; i < timeLsResultLines; i++ {
 							line := ublx.UbloxPollPull()
 							lines = append(lines, line)
 						}
 						timeLs := ublox.ExtractLeapSec(lines)
-						g.leapManager.UbloxLsInd <- *timeLs
+						select {
+						case g.leapManager.UbloxLsInd <- *timeLs:
+						case <-time.After(100 * time.Millisecond):
```

Comment:
Can you break this out of the inner for loop entirely?  Ideally after the event code below.  Otherwise, the reported information from the NAV-CLOCK and NAV-STATUS could be a bit old.  Also only want to do it once, not multiple times if there have been multiple of them in the ublox output.  Code at line 268 is to make sure we only look at the last occurrence of this.